pkg/addon/controllers/addonprogressing/controller.go

```diff
@@ -285,18 +285,6 @@ func workIsReady(work *workapiv1.ManifestWork) bool {
 
 // set addon progressing condition and last applied
 func setAddOnProgressingAndLastApplied(reason string, message string, addon *addonapiv1alpha1.ManagedClusterAddOn) {
-	// always update progressing condition when there is no config
```

Comment:
why do we need to remove this skip?

Reply:
Meet an issue: when addon applied configs successfully and then configs become unaccessible (eg, removed), the desired spec hash will be empty and report "mca and work configs mismatch", and when the configs added back, since lastapplied already the same as desired, so the error message will leave in the condition and can never trigger an update. 

2 reasons feel save to remove the skip: 
- In the previous code, add a skip to avoid "install" message changes to "upgrade" after a fresh install. With PR https://github.com/open-cluster-management-io/ocm/pull/499/files merged, won't have such issue. 
- Another reason is once the mca applied config successfully (lastapplied = desired), do not want to change the "Progressing" condition any more, to avoid back and forth in rollout. Feel save to remove the skip because the [setRolloutStatus](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/addon/controllers/addonconfiguration/graph.go#L54) only look at the lastapplied and desired, not message.

Reply:
could we add some comments here?

Reply:
added.